trending_data/src/main/webapp/unique_window_handler.js

```diff
@@ -41,22 +31,24 @@ export class UniqueWindowHandler {
   }
 
   /**
-  * Called when Twitter button is clicked. Clears
+  * Called when YouTube button is clicked. Clears
   * current content of window and shows the most recently
-  * cached Twitter data
+  * cached YouTube data
   */
   showYTData() {
+    this.showing = 'yt';
     this.initDataWindow(); // clear current content and re-add buttons
     this.dataWindow.appendChild(this.ytDataDiv);
     this.currentWindow.setContent(this.dataWindow);
   }
```

Comment:
If someone clicks the button while already showing youtube. It unnecessarily loads the page again.

Reply:
It does not load it again if user clicks on the same marker while the window is open

Reply:
If the window is already open and showing youtube and someone clicks on the youtube button again, it will reload the content of the page while ideally nothing should happen, it's just extra unnecessary work done.

Reply:
It just appends the already populated `ytDataDiv`, data is not fetched again

Reply:
If yt button is clicked -> showYTData()
  - clears current data
  - shows cached yt data

`ytDataDiv` is populated when the window is open (`this.loadYTData()` inside `openWindow`) so the content is saved and can be reused in the current window

Reply:
I know data is not fetched again but the window is loaded again. Try opening a window and clicking on youtube multiple times and you'll see what I mean.

Reply:
Also I don't know if this is intentional but the buttons lost their colors.

Reply:
Yes I just fixed that bug

Reply:
And fixed the other problem where window is reinitialized if click on same button. Now when button is clicked we check if we're already showing the requested data.